src/tape/ensemble.py

```diff
@@ -1142,6 +1160,65 @@ def s2n_inter_quartile_range(flux, err):
                 meta=meta,
             )
 
+        # Output standardization
+        if isinstance(batch, EnsembleSeries):
+            if batch.name == self._id_col:
+                batch = batch.rename("result")
+            batch = EnsembleFrame.from_dask_dataframe(batch.to_frame())
+            if len(on) > 1:
+                batch = batch.reset_index()
+                # Need to overwrite the meta manually as the multiindex will be
+                # interpretted by dask as a single "index" column
+                batch._meta = TapeFrame(columns=on + ["result"])
+
+                if by_band:
+                    batch = batch.categorize(self._band_col).pivot_table(
+                        index=on[0], columns=self._band_col, aggfunc="sum"
+                    )
+
+                    # Need to once again reestablish meta for the pivot
+                    band_labels = batch.columns.values
+                    out_cols = []
+                    for col in ["result"]:
+                        for band in band_labels:
+                            out_cols += [(str(col), str(band))]
+                    batch._meta = TapeFrame(columns=out_cols)
+                    # Flatten the columns to a new column per band
+                    batch.columns = ["_".join(col).strip() for col in batch.columns.values]
+
+                    # The pivot returns a dask dataframe, need to convert back
+                    batch = EnsembleFrame.from_dask_dataframe(batch)
+
+                else:
+                    batch = batch.set_index(on[0], sort=False)
+
+        elif isinstance(batch, EnsembleFrame):
+            if len(on) > 1:
+                res_cols = list(batch._meta.columns)
+                batch = batch.reset_index()
+                batch._meta = TapeFrame(columns=on + res_cols)
+                if by_band:
+                    batch = batch.categorize(self._band_col)
+                    batch = batch.pivot_table(index=on[0], columns=self._band_col, aggfunc="sum")
+
+                    # Need to once again reestablish meta for the pivot
+                    band_labels = batch.columns.values
+
+                    out_cols = []
+                    for col in res_cols[::-1]:
+                        for band in band_labels:
+                            out_cols += [(str(col), str(band))]
+                    batch._meta = TapeFrame(columns=out_cols)
+
+                    # Flatten the columns to a new column per band
+                    batch.columns = ["_".join(col).strip() for col in batch.columns.values]
+
+                    # The pivot returns a dask dataframe, need to convert back
+                    batch = EnsembleFrame.from_dask_dataframe(batch)
+
+                else:
+                    batch = batch.set_index(on[0], sort=False)
+
```

Comment:
Do we need to check some third case after `isinstance(batch, EnsembleSeries)` and `isinstance(batch, EnsembleFrame)`?

Reply:
Those should encompass the only real outcomes, but maybe good to add an else that raises an error just for completeness.

Reply:
addressed in latest commit